src/main/java/lotto/domain/LottoTicketMachine.java

```diff
@@ -2,35 +2,67 @@
 
 import lotto.dto.LottoStatisticsDto;
 
+import java.util.List;
+
 public class LottoTicketMachine {
 
     public static final int LOTTO_PRICE = 1000;
 
-    private LottoTickets lottoTickets;
-    private int lottoCount;
+    public LottoTickets lottoTicketsTotal;
+    public int autoPurchaseCount;
+
+    public LottoTicketMachine(int manualPurchaseCount, int purchaseAmount) {
+        validateManualPurchaseCount(manualPurchaseCount, purchaseAmount);
+        this.autoPurchaseCount
+                = calculateAutoPurchaseCount(manualPurchaseCount, purchaseAmount);
+    }
+
+    private int calculateAutoPurchaseCount(int manualPurchaseCount, int purchaseAmount) {
+        return (purchaseAmount - LOTTO_PRICE * manualPurchaseCount) / LOTTO_PRICE;
+    }
+
+    private void validateManualPurchaseCount(int manualPurchaseCount, int purchaseAmount) {
+        if (purchaseAmount < manualPurchaseCount * LOTTO_PRICE) {
+            throw new IllegalArgumentException("구입할 금액을 초과합니다.");
+        }
+    }
+
+    public void createLottoTickets(List<List<Integer>> tickets) {
+        if(autoPurchaseCount == 0) {
+            createManualLottoTickets(tickets);
+            return;
+        }
+        if (tickets.size() == 0) {
```

Comment:
`List`의 `isEmpty()`를 활용해봐도 좋을 것 같아요.
```suggestion
        if (tickets.isEmpty()) {
```